File: task-03/Haskell/code4.hs
import System.IO

printDiamond :: Int -> String
printDiamond n = unlines (upperHalf ++ lowerHalf)
  where
    upperHalf = [replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*' | i <- [0..n-1]]
    lowerHalf = [replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*' | i <- reverse [0..(n-2)]]

main :: IO ()
main = do
    content <- readFile "input.txt"
    let n = read (init content) :: Int
    let diamond = printDiamond n
    writeFile "output.txt" diamond

File: task-03/Java/code4.java
import java.nio.file.Files;
import java.nio.file.Paths;
import java.io.IOException;
import java.util.stream.IntStream;

public class DiamondPattern {
    public static void main(String[] args) {
        try {
            String content = new String(Files.readAllBytes(Paths.get("input.txt")));
            int n = Integer.parseInt(content.trim());
            
            StringBuilder diamond = new StringBuilder();
            
            for (int i = 0; i < n; i++) {
                diamond.append(" ".repeat(Math.max(0, n - i - 1)))
                        .append("*".repeat(2 * i + 1))
                        .append("\n");
            }
            for (int i = n - 2; i >= 0; i--) {
                diamond.append(" ".repeat(Math.max(0, n - i - 1)))
                        .append("*".repeat(2 * i + 1))
                        .append("\n");
            }
            
            Files.write(Paths.get("output.txt"), diamond.toString().getBytes());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}